packages/playwright-core/src/server/trace/recorder/snapshotterInjected.ts

```diff
@@ -337,8 +336,6 @@ export function frameSnapshotStreamer(snapshotStreamer: string) {
           if (rel === 'preload' || rel === 'prefetch')
             return;
         }
-        if (this._removeNoScript && nodeName === 'NOSCRIPT')
```

Comment:
I don't think this will help by itself. Trace viewer does not render the snapshot with javascript disabled, so the browser will not show the `noscript` tag contents.

Reply:
> Trace viewer does not render the snapshot with javascript disabled, so the browser will not show the noscript tag contents.

If you look at the actual test results, you'll immediately see, that it seems to work in practice! :)

well, the semantic of the `noscript` tag is rather complex, and I don't know, if this very simple modification catches all the possible corner cases, but at least for very basic usage scenarios it seems to work.

Reply:
Well, I tried it myself, and in the trace viewer I see (despite `<noscript>` tag present):
```txt
Here you should see a message, if JS was dissabled durring this Test:

""
This output should look identical:

"JS Disabled (2)"
```

while during test execution I see:

```txt
Here you should see a message, if JS was dissabled durring this Test:

"JS Disabled (1)"
This output should look identical:

"JS Disabled (2)"
```

So, I don't think it helps meaning that trace viewer will not show the snapshot as it was visible during the test itself.

Reply:
If my fix isn't enabled and JS disabled, you see the fist output, otherweise the second one.  
if JS is enabled, both strings are empty.

All browser handle this kind of test in a correct manner. They even send correct results to the tracer! It's really only the incorrect code in playright, which is responsible for the wrong results.

Reply:
@dgozman 
I just looked at the test myself again to meditate about your claim.

what i saw:
If hover over the timeline, i get the correct result resp. the expected output after applying my patch and somthing that corresponds very well to the test code results.
**but**
the main view of the tracer still looks different! :(

![Screenshot from 2023-10-13 01-51-06](https://github.com/microsoft/playwright/assets/13013641/cea88cc6-c5da-4e8f-83d9-9492c7d1ff28)
![Screenshot from 2023-10-13 01-50-29](https://github.com/microsoft/playwright/assets/13013641/7a90faee-6ebc-4931-8ec6-be5210efaff3)

Reply:
> If hover over the timeline, I get the correct result resp. the expected output after applying my patch and something that corresponds very well to the test code results.

Yep, that's what the browser renders.

> the main view of the tracer still looks different! :(

Exactly! This is rendered in the trace viewer from the snapshot that you are changing. That's what I mean by "this change will not help the trace viewer", because it will effectively ignore the `<noscript>` entirely.

Reply:
I send you a working patch soon.

but I still have to find a nice way to transfer the javascriptEnabled context state into this method.

And I additionally had to mask Noscript tags in the trace viewer. Otherwise they will be interpreted by the display mechanism there again.